src/_components/AssetDetails/AssetDetailsContainer.jsx

```diff
@@ -0,0 +1,38 @@
+import { connect } from 'react-redux';
+
+import { getAssetDetail } from '../../_actions/asset.actions';
+import { loadAssetAssigneeUsers } from '../../_actions/users.actions';
+import AssetDetailsComponent from '../../components/AssetDetails/AssetDetailsComponent';
+
+export const isSameId = (assetDetail, props) => (assetDetail.uuid === props.match.params.id);
+
+export const mapStateToProps = ({ asset, usersList }, ownProps) => {
+  const {
+    assetDetail,
+    errorMessage,
+    hasError,
+    newAllocation,
+    unAssignedAsset,
+    buttonLoading
+  } = asset;
+  const { assetAsigneeUsers } = usersList;
+  const assetLoading = asset.isLoading;
+  const userLoading = usersList.isLoading;
+
+  return {
+    assetAsigneeUsers,
+    assetDetail,
```

Comment:
`assetDetail` will always be `{}` Or _have the previous data_. When you click an asset row in `/assets` page, the asset detail for that asset will be passed in `props.location.state`. This means that you won't need to make an API request, even if `isSameId` evaluates to false. This is the reason why I had written the following in one  of my previous comment:
```jsx
const getAssetDetail = (assetDetail, props) => {
  const hasSameId = isSameId(assetDetail, props);
  if (hasSameId) {
    return assetDetail;
  }
  return get(props.location, 'state', {});
}
```
If you have changed the logic because the data is not saved in store when the info is obtained from location props, then you need to add it to sore. For example, in the container you can have the following:
```jsx
import { getAssetDetail, createAssetSuccess as addAsset } from '../../_actions/asset.actions';

const mapStateToProps = ({ asset, usersList }, ownProps) => {
  const {assetDetail} = asset;
  const details = getAssetDetail(assetDetail, ownProps);
  const hasSameId = isSameId(assetDetail, ownProps);
  const shouldFetchDetails =  isEmpty(details);

  // if the id is different and `shouldFetchDetails` is false, then you know you should add the info to the store
  const shouldAddToStore = !hasSameId && !shouldFetchDetails;

  return {
    shouldFetchDetails,
    assetDetails: details,
    shouldAddToStore,
    ...
    buttonLoading
  }
}

export default connect(mapStateToProps, {
  getAssetDetail,
  loadAssetAssigneeUsers,
  addAsset
})(AssetDetailsComponent);
```
Then in `componentDidmount` of the component, you'll do the following:
```jsx
componentDidMount() {
  if (shouldFetchDetails) {
    this.props.getAssetDetail(id);
  }
  if (shouldAddToStore) {
    this.props.addAsset(assetDetail)
  }
}
```
What we are trying to avoid is making API requests to fetch data that already exist somewhere else. Therefore, you have to use the data from `props.location.state`

Reply:
Okay. I actually changed it coz it wasn't adding the data to the store. I'll work on the changes you've suggested

Reply:
Hey @sachungo 
When I implement the above feedback, a bunch of duplicates are created every time I view an asset as seen below.

<img width="1335" alt="screenshot 2018-11-09 at 09 03 42" src="https://user-images.githubusercontent.com/31339738/48245970-95d89e80-e3fe-11e8-9c02-d957dea934bb.png">

These duplicates, however, come without an assigned user.